### PR TITLE
Add an armadillo

### PR DESCRIPTION
**Pull request by IProduceWidgets**
## About the PR

Sprites by imasleeping

add an armadillo
image

It turns into a ball when it goes into crit.
It makes a funny noise.
It has some slight armor to slash and pierce, and a slight vulverability to heat.

Also fixed some small mistakes in the naming of other cargo livestock crate entities.

## Why / Balance

More zookeeper content real.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: imasleeping, IProduceWidgets
- add: an Armadillo!

### CODE REVIEW

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 1]
```diff
@@ -2749,3 +2749,61 @@
...
+  - type: Tag
+    tags:
+    - DoorBumpOpener
+    - VimPilot
+  - type: Armor # its got a shell, I guess.
+    modifiers:
+      coefficients:
+        Slash: 0.90
+        Piercing: 0.90
+        Heat: 1.10
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2809]: this doesn't actually do anything. You'd need a damage modifier set.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 2]
```diff
@@ -2749,3 +2749,61 @@
...
+  name: armadillo
+  parent: SimpleMobBase
+  id: MobArmadillo
+  description: some sort of armored rat.
+  components:
+  - type: MovementSpeedModifier
+    baseWalkSpeed: 4
+    baseSprintSpeed: 6
+  - type: Sprite
+    drawdepth: Mobs
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2762]: you can drop the drawdepth line

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 3]
```diff
@@ -2749,3 +2749,61 @@
     factions:
     - Passive
 
+- type: entity
+  name: armadillo
+  parent: SimpleMobBase
+  id: MobArmadillo
+  description: some sort of armored rat.
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2756]:
```suggestion
  description: Some sort of armored rat.
```

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 4]
```diff
@@ -2749,3 +2749,61 @@
...
+  - type: MovementSpeedModifier
+    baseWalkSpeed: 4
+    baseSprintSpeed: 6
+  - type: Sprite
+    drawdepth: Mobs
+    layers:
+    - map: ["enum.DamageStateVisualLayers.Base"]
+      state: armadillo
+      sprite: Mobs/Animals/armadillo.rsi
+  - type: Physics
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2767]: unnecessary

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 5]
```diff
@@ -2749,3 +2749,61 @@
...
+        mask:
+        - MobMask
+        layer:
+        - MobLayer
+  - type: DamageStateVisuals
+    states:
+      Alive:
+        Base: armadillo
+      Critical:
+        Base: ball
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2784]: the crit state being a ball is kinda strange. If we have support for walk animations, then the ball rolling would be good. I'm not too sure where to look for that, however.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml (hunk 5 above)
Comment (IProduceWidgets) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2784]: I was unable to find it if it exists, that was my original idea.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml (hunk 1 above)
Comment (IProduceWidgets) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2809]: Ahh thanks!

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml (hunk 3 above)
Comment (IProduceWidgets) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml]: Done! I made a github boo boo though, sorry for the re-review!

**Comment by HerCoyote23** [comment]
I say give it more armor! It's got almost none. I know that's realistic, but I think it'd be cooler to have a properly armored armadillo.
That, or add a rare variant of it called an "Armordillo" that has 95% resist to everything.

**Comment by lzk228** [comment]
what is it? a minecraft? :trollface:

File: Resources/Audio/Animals/attributions.yml [hunk 6]
```diff
@@ -111,4 +111,9 @@
 - files: ["kangaroo_grunt.ogg"]
   license: "CC-BY-4.0"
   copyright: "Audio is recorded/created by Pfranzen 'FreeSound.org'. The original audio was trimmed and renamed"
-  source: "https://freesound.org/people/pfranzen/sounds/322744/"
\ No newline at end of file
+  source: "https://freesound.org/people/pfranzen/sounds/322744/"
+
+- files: ["armadillo_scream.ogg"]
+  license: "CC-BY-3.0"
```
Comment (TheShuEd) [inline Resources/Audio/Animals/attributions.yml:117]: I'm not sure if youtube videos have a free license. It's worth double-checking.

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml [hunk 7]
```diff
@@ -197,3 +197,13 @@
   cost: 2800
   category: Livestock
   group: market
+
+- type: cargoProduct
+  id: LivestockArmadillo
+  icon:
+    sprite: Mobs/Animals/kangaroo.rsi
+    state: kangaroo
```
Comment (TheShuEd) [inline Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:205]: kangaroo?

File: Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl [hunk 8]
```diff
@@ -53,4 +53,7 @@ ent-CrateNPCLizard = Lizard crate
     .desc = A crate containing a lizard.
 
 ent-CrateNPCKangaroo = Kangaroo crate
-    .desc = A crate containing a kangaroo.
\ No newline at end of file
+    .desc = A crate containing a kangaroo.
+
+ent-CrateNPCArmadillo = Armadillo crate
+    .desc = A crate containing an armadillo.
```
Comment (TheShuEd) [inline Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl:59]: "an" while all others are signed as "a". Scheduled?

File: Resources/Textures/Mobs/Animals/armadillo.rsi/meta.json [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  },
+  "license": "CC-BY-SA-3.0",
+  "copyright": "created by imasleeping (Discord) and IProduceWidgets (github)",
+  "states": [
+    {
+      "name": "armadillo",
+      "directions": 4,
+      "delays": [
+        [
+          1
```
Comment (TheShuEd) [inline Resources/Textures/Mobs/Animals/armadillo.rsi/meta.json:15]: no need to prescribe animation delays if we have only 1 frame

File: Resources/Textures/Mobs/Animals/armadillo.rsi/meta.json [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        ]
+      ]
+    },
+	    {
+      "name": "ball",
+      "directions": 4,
+      "delays": [
+        [
+          1
```
Comment (TheShuEd) [inline Resources/Textures/Mobs/Animals/armadillo.rsi/meta.json:33]: above

File: Resources/Textures/Mobs/Animals/armadillo.rsi/meta.json [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        ]
+      ]
+    },
+    {
+      "name": "dead",
+      "delays": [
+        [
+          1
```
Comment (TheShuEd) [inline Resources/Textures/Mobs/Animals/armadillo.rsi/meta.json:50]: above

**Review by TheShuEd — CHANGES_REQUESTED** [review]

File: Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl (hunk 8 above)
Comment (deltanedas) [reply Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl:59]: armadillo starts with a so an

Comment (TheShuEd) [reply Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl:59]: Oh, okay, I shouldn't have gotten into English grammar. :trollface:

**Comment by IProduceWidgets** [comment]
Converted to draft because I want to use the new functionality for walking sprites

**Comment by metalgearsloth** [comment]
Does this mob offer any gameplay or is it just a reskin?

**Comment by IProduceWidgets** [comment]
It shoots some waterguns, but is otherwise just an animal for a watery area.

**Comment by AJCM-git** [comment]
New fluff animals ordered from cargo or spawned in the station have been added to the freeze list #8524 with new requirements for they to be approved (tldr is that they must have something cool going on apart from being cute and having sprites, like new AI behaviors or being spawned in proc gens maps), feel free to re-open if you meet them.

**Pull request closed without merging**